test/player_unit.js

```diff
@@ -703,6 +704,39 @@ describe('Player', () => {
       expect(seekRange.end).toBe(10);
     });
 
+    // Test for https://github.com/shaka-project/shaka-player/issues/4026
+    it('configures play and seek range with notifySegments', async () => {
+      player.configure({playRangeStart: 5, playRangeEnd: 10});
+      const timeline = new shaka.media.PresentationTimeline(300, 0);
+      timeline.setStatic(true);
+      timeline.setDuration(300);
```

Comment:
It's a bit confusing how in this test, `playRangeEnd` corresponds with `duration_` inside the timeline, and the `setDuration` method call corresponds with `maxSegmentEndTime_` in the timeline. Feels like the two things named duration should match up.
That's out of the scope of this CL though.

Reply:
Yeah, it's weird.  What's really happening is that stream.useSegmentTemplate refers to timeline's duration to decide how many references to generate.  Then player.load applies the config to the timeline, and sets the duration lower.

What would be less confusing, I think, is if Timeline had a separate field for the playRangeEnd setting.  But that would also require a lot more ifs or mins in Timeline, which could be confusing in a different way.